### PR TITLE
redis: add_command method & small rename_commands fix

### PR DESCRIPTION
**Pull request by 43081j**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/NodeRedis/node_redis (README)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

---

This adds `add_command` which allows you to.. add a command. introduced in 2.8.

It also makes a slight change to `rename_commands` in that, as per the docs, it can take a value of `null`. This means the resulting type is actually `{[command: string]: string} | null | undefined` pretty much.

### CODE REVIEW

File: types/redis/index.d.ts [hunk 1]
```diff
@@ -43,7 +44,7 @@ export interface ClientOpts {
     password?: string;
     db?: string | number;
     family?: string;
-    rename_commands?: { [command: string]: string };
+    rename_commands?: { [command: string]: string } | null;
```
Comment (Rokt33r) [inline types/redis/index.d.ts]: `rename_commands` is already set optional. Why do we need this?

**Review by Rokt33r — CHANGES_REQUESTED** [review]

File: types/redis/index.d.ts (hunk 1 above)
Comment (43081j) [reply types/redis/index.d.ts]: my mistake, i was under the assumption we had `strictNullChecks` turned on

**Comment by 43081j** [comment]
Regarding the one comment we have here, see [here in the README](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#some-packages-have-no-tslintjson-and-some-tsconfigjson-are-missing-noimplicitany-true-noimplicitthis-true-or-strictnullchecks-true).

I'm tempted to update our tsconfig in this case...

**Comment by 43081j** [comment]
I've added `strictNullChecks: true` as per the [README](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#some-packages-have-no-tslintjson-and-some-tsconfigjson-are-missing-noimplicitany-true-noimplicitthis-true-or-strictnullchecks-true).

I've had a quick look through what this changes... basically the only things to think about are the optional settings. With this change, most of them cannot be `null` anymore. But, having looked through the source, i don't see why you would ever pass null so the current types should be fine.

If you disagree, do let me know why.

**Comment by Rokt33r** [comment]
It seems to be right. I agree with you.

**Review by Rokt33r — APPROVED** [review]

**Comment by OmgImAlexis** [comment]
This commit is wrong the method isn't on the client and never was https://github.com/NodeRedis/node-redis/commit/0437aa49851365bdf82711b7ff6ac7dc369ec7bb

**Pull request merged**